classes/task/get_meeting_recordings.php

```diff
@@ -109,17 +104,29 @@ public function execute() {
             $zoomrecordings = $service->get_user_recordings($hostid, $from, $to);
 
             foreach ($zoomrecordings as $recordingid => $recording) {
-                if (isset($localrecordings[$recording->meetinguuid][$recordingid])) {
-                    mtrace('Recording id: ' . $recordingid . ' exists...skipping');
-                    continue;
-                }
-
                 if (empty($meetings[$recording->meetingid])) {
                     // Skip meetings that are not in Moodle.
                     mtrace('Meeting id: ' . $recording->meetingid . ' does not exist...skipping');
                     continue;
                 }
 
+                $zoom = $meetings[$recording->meetingid];
+
+                if (isset($localrecordings[$recording->meetinguuid][$recordingid])) {
+                    mtrace('Recording id: ' . $recordingid . ' exists...verifying current data.');
+                    $localrecording = $localrecordings[$recording->meetinguuid][$recordingid];
+
+                    if ($localrecording->name !== $zoom->name) {
+                        $updatemeeting = (object) [
+                            'id' => $localrecording->id,
+                            'name' => $zoom->name,
+                        ];
+                        $DB->update_record('zoom_meeting_recordings', $updatemeeting);
+                    }
+
```

Comment:
The get_meeting_recordings task only has a one day lookback. So it's unlikely that we're going to get any benefit from this. If anything, the "update the name" code would be better as part of a "fix the existing data" tool that gets run once, manually by an administrator.

```suggestion
                    mtrace('Recording id: ' . $recordingid . ' exists...skipping');
```

Reply:
All future content should be correct, so the problem is existing content. Recordings do not have their own in the Zoom API, so setting the recording->name to the meeting->name as part of the upgrade.php step would fix all of the entries that we wanted this code to fix in the first place, without adding overhead to regularly scheduled processes.